sys/parsoid.js

```diff
@@ -630,29 +626,18 @@ class ParsoidService {
         const rp = req.params;
         const tid = uuid.now().toString();
         const wtType = req.original && req.original.headers['content-type'] || 'text/plain';
-        return transformPromise.then((original) =>
-            // Save the returned data-parsoid for the transform and the wikitext sent by the client
-            P.all([
-                hyper.put({
-                    uri: this.getStashBucketURI(rp, 'data-parsoid', tid),
-                    headers: original.body['data-parsoid'].headers,
-                    body: original.body['data-parsoid'].body
-                }),
-                hyper.put({
-                    uri: this.getStashBucketURI(rp, 'wikitext', tid),
+        return transformPromise.then((original) => hyper.put({
+            uri: this.getStashBucketURI(rp.domain, rp.title, rp.revision, tid),
+            body: Buffer.from(JSON.stringify({
```

Comment:
Also for the sake of consistency, perhaps we should create a `mwUtils.encodeBody()` function that encodes this, and use it in the mobile sections part too.

Reply:
Idem